portal.go

```diff
@@ -1754,6 +1754,49 @@ func (portal *Portal) GetMatrixUsers() ([]id.UserID, error) {
 	return users, nil
 }
 
+// TombstoneOrReIDIfNeeded returns true if the portal metadata should be synchronized
+func (portal *Portal) TombstoneOrReIDIfNeeded() bool {
+	if portal.Identifier.Service == "SMS" && portal.bridge.Config.IMessage.TombstoneOldRooms {
+		if len(portal.MXID) == 0 {
+			return false
+		}
+		identifier := portal.Identifier
+		identifier.Service = "iMessage"
+		if replacement := portal.bridge.DB.Portal.GetByGUID(identifier.String()); replacement != nil {
+			if len(replacement.MXID) == 0 {
+				portal.log.Infofln("ReID %s to %s for chat merging", portal.Identifier.String(), identifier.String())
+				replacement.Delete()
+				portal.ReID(identifier.String())
+				portal.Identifier = identifier
+				return true
+			}
+			var tombstoneContent event.TombstoneEventContent
+			if err := portal.MainIntent().StateEvent(portal.MXID, event.StateTombstone, "", &tombstoneContent); err != nil {
+				portal.log.Errorfln("Error while getting tombstone state event in portal %s: %v", portal.MXID, err)
+				return false
+			}
+			if len(tombstoneContent.ReplacementRoom) == 0 {
```

Comment:
This check can be removed now that the portal is being deleted (and it won't try to re-merge on the next startup)